src/System.Memory/tests/Performance/Perf.Utf8Parser.cs

```diff
@@ -10,12 +11,609 @@ namespace System.Buffers.Text.Tests
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        int.TryParse(s_Int32TextArray[i % textLength], out int value);
+                        TestHelpers.DoNotIgnore(value, 0);
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        [InlineData("0")]
+        [InlineData("10737")] // standard parse
+        [InlineData("32767")] // max value
+        [InlineData("-32768")] // min value
+        [InlineData("000000000000000000001235abcdfg")]
+        [InlineData("214abcdefghijklmnop")]
+        [InlineData("-2147abcdefghijklmnop")]
+        [InlineData("21474abcdefghijklmnop")]
+        private static void ByteSpanToInt16(string text)
+        {
+            byte[] utf8ByteArray = Encoding.UTF8.GetBytes(text);
+            var utf8ByteSpan = new ReadOnlySpan<byte>(utf8ByteArray);
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        Utf8Parser.TryParse(utf8ByteSpan, out short value, out int consumed);
+                        TestHelpers.DoNotIgnore(value, consumed);
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        private static void ByteSpanToInt16_VariableLength()
+        {
+            int textLength = s_Int16TextArray.Length;
+            byte[][] utf8ByteArray = (byte[][])Array.CreateInstance(typeof(byte[]), textLength);
+            for (var i = 0; i < textLength; i++)
+            {
+                utf8ByteArray[i] = Encoding.UTF8.GetBytes(s_Int16TextArray[i]);
+            }
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        ReadOnlySpan<byte> utf8ByteSpan = utf8ByteArray[i % textLength];
+                        Utf8Parser.TryParse(utf8ByteSpan, out short value, out int bytesConsumed);
+                        TestHelpers.DoNotIgnore(value, bytesConsumed);
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        [InlineData("0")]
+        [InlineData("10737")] // standard parse
+        [InlineData("32767")] // max value
+        [InlineData("-32768")] // min value
+        [InlineData("-127")]
+        [InlineData("2147")]
+        [InlineData("000000000000000000001235")]
+        private static void ByteSpanToInt16_Baseline(string text)
+        {
+            byte[] utf8ByteArray = Encoding.UTF8.GetBytes(text);
```

Comment:
We don't use the utf8ByteArray or utf8ByteSpan locals anywhere here and can be removed.